src/sagemaker/jumpstart/utils.py

```diff
@@ -1595,9 +1595,10 @@ def _add_model_access_configs_to_model_data_sources(
             )
             acked_model_data_sources.append(mutable_model_data_source)
         else:
-            mutable_model_data_source.pop(
-                "HostingEulaKey"
-            )  # pop when model access config is not applicable
+            if "HostingEulaKey" in mutable_model_data_source:
```

Comment:
Can we add a test for this in `test_model_builder` as well in addition to what you already have ? This is a very critical path in ModelBuilder

Reply:
[tests/integ/sagemaker/serve/test_serve_js_deep_unit_tests.py](https://github.com/aws/sagemaker-python-sdk/pull/4937/files#diff-557ffe7eb4dcf1f09701ae8c60fbdb57e8d115913c2e5d4f6c5ad9de641a68aa) under `test_js_model_with_optimize_speculative_decoding_config_gated_requests_are_expected`